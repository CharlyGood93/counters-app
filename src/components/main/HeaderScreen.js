import React from 'react';
import { Input, SearchIcon } from '../../ui';

export const HeaderScreen = () => {
    return (
        <section className="main__searchbar">
            <div className="main__searchbar-icon">
                <SearchIcon className="icon" />
                <Input className="main__searchbar-input" placeholder="Search Counters" />
            </div>
        </section>
    )
}
