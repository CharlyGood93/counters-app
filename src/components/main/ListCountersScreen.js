import React, { useState } from 'react';
import { Col, Container, ListGroup, Row } from 'react-bootstrap';
import { Button, DecrementIcon, IncrementIcon } from '../../ui';

export const ListCountersScreen = ({ products }) => {

    const [counter, setCounter] = useState();

    const handleDecrementCount = (count) => {
        console.log('Decrement counters');
    }

    const handleIncrementCount = (count) => {
        console.log('Increment counters');
    }

    return (
        <Container fluid>
            <Row style={{ alignContent: 'center' }}>
                <Col>
                    {
                        products.length > 1 ? <><strong>{products.length} items</strong></> : <>{products.length} item</>
                    }
                </Col>
                <Col>
                    {
                        products.reduce((a, b) => a + b.count, 0)
                    }
                    &nbsp;times
                </Col>
                <Col>
                    Icon Refresh
                </Col>
            </Row>
            <ListGroup>
                {
                    products.map((product, i) => (
                        <ListGroup.Item key={product.id}>
                            <Row>
                                <Col>
                                    {product.title}
                                </Col>
                                <Col>
                                    <Button size="big" color="white">
                                        <DecrementIcon fill="var(--app-tint)" />
                                    </Button>
                                    {product.count}
                                    <Button size="big" color="white">
                                        <IncrementIcon fill="var(--app-tint)" />
                                    </Button>
                                </Col>
                            </Row>
                        </ListGroup.Item>
                    ))
                }
            </ListGroup>
        </Container>
    )
}
