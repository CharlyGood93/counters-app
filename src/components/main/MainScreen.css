.main {
    padding: 1rem;
}

.main__footer {
    border-top: 1px solid var(--silver);
    bottom: 0;
    left: 1rem;
    padding: 1rem;
    position: fixed;
    right: 1rem;
    text-align: center;
}

.main__footer .main__footer-add {
    float: right;
}

.main .main__loading {
    left: 50%;
    position: fixed;
    top: 50%;
    transform: translate(-50%, -50%);
}

.main__modal-header-title {
    display: flex;
    align-items: center;
}

.main .main__no-counters {
    margin-top: 200px;
    text-align: center;
}

.main .main__no-counters-cite, .main .main__no-counters-subtitle {
    color: var(--grey);
}

.main .main__no-counters-title {
    color: var(--dark-black);
}

.main__searchbar {
    border-bottom: 1px solid var(--silver);
}

.main__searchbar-icon svg {
    position: absolute;
    left: 35px;
    top: 30px;
}

.main__searchbar-icon {
    margin-bottom: 1rem;
    width: 100%;
}

.main__searchbar-input {
    width: 100%;
    padding-left: 50px;
    text-align: left;
}