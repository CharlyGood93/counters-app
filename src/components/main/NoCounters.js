import React from 'react'

export const NoCounters = () => {
    return (
        <section className="main__no-counters">
            <h3 className="main__no-counters-title">
                <strong>No counters yet</strong>
            </h3>
            <p className="main__no-counters-subtitle">
                <q>When I started couting my blessings, my whole life turned around.</q>
            </p>
            <p className="main__no-counters-cite">
                &mdash; Willie Nelson
            </p>
        </section>
    )
}
