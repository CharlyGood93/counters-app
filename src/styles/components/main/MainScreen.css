.container-fluid {
    margin-top: auto;
    margin-bottom: auto;
    padding-top: 15px;
    padding-bottom: 15px;
}

.container-fluid__no-counters {
    margin: 150px 0 150px 0;
}

.container-fluid__no-counters h4 {
    color: var(--dark-black);
}

.container-fluid__no-counters p {
    color: var(--grey);
}

.footer {
    border-top: 1px solid var(--silver);
    bottom: 0;
    height: 60px;
    line-height: 60px;
    position: absolute;
    width: 100%;
}

.footer__add-counter Button {
    float: right;
}

.footer__actions Button, .footer__add-counter Button {
    margin: 5px;
}

.footer__actions Button span svg, .footer__add-counter Button span svg, .modal__close-button Button span svg {
    margin: 0px;
}