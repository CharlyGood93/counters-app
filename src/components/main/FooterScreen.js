import React, { useEffect, useState } from 'react';
import { ExampleScreen } from '../examples/ExampleScreen';
import { Button, CloseIcon, Input, Loading, Modal, NewIcon, useModal } from '../../ui';
import { postNewCounters } from '../../api/postNewCounters';

import './MainScreen.css';

export const FooterScreen = () => {


    const [showExamples, setShowExamples] = useState(false);

    const handleViewExamples = () => {
        setShowExamples(true);
    }

    const { isVisible: isModalVisible, hideModal, showModal } = useModal();

    const showOrHideModalBody = () => {
        if (showExamples) {
            setShowExamples(false);
        } else {
            hideModal(true);
        }
    }

    const [countersValue, setCountersValue] = useState('');

    const handleInputChange = (e) => {
        e.preventDefault();
        setCountersValue(e.target.value);
    }

    const [newCounters, setNewCounters] = useState();

    const handleAddCounters = (e) => {
        e.preventDefault();
        console.log('Your submited me!');
        console.log({ countersValue });
        if (countersValue !== '') {
            console.log('ahora puedo guardar');
            const saveNewCounter = postNewCounters(countersValue);
            saveNewCounter.then(resp => {
                console.log(resp.data);
                setNewCounters(resp.data);
            });
            console.log(newCounters);
            setCountersValue('');
        }
    }

    // postNewCounters();

    return (
        <>
            <section className="main__footer">
                <Button
                    className="main__footer-add"
                    onClick={showModal}>
                    <NewIcon fill="white" />
                </Button>
            </section>
            <Modal
                isVisible={isModalVisible}>
                <Modal.Header>
                    <Modal.Title className="main__modal-header-title">
                        <Button
                            color="grey"
                            onClick={showOrHideModalBody}
                            size="big">
                            <CloseIcon fill="white" />
                        </Button>
                        {
                            showExamples
                                ?
                                <p>Examples</p>
                                :
                                <>
                                    <p>Create counter</p>
                                    <Button onClick={handleAddCounters}>
                                        Save
                                    </Button>
                                </>
                        }
                    </Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    {
                        showExamples ?
                            <ExampleScreen />
                            :
                            <>
                                <section className="main__loading">
                                    {/* <Loading /> */}
                                </section>
                                <p>
                                    <strong>
                                        Name
                                    </strong>
                                </p>
                                <Input onChange={handleInputChange} placeholder="Cups of coffee" type="text" value={countersValue} />
                                <p>
                                    Give it a name. Creative block?
                                    See <u style={{ cursor: 'pointer' }} onClick={handleViewExamples}>examples</u>.
                                </p>
                            </>
                    }
                </Modal.Body>
            </Modal>
        </>
    )
}
