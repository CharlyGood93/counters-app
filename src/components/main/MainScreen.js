import React, { useEffect, useState } from 'react';

import { getAllCounters } from '../../api/getAllCounters';

import { Col, Container, Row } from 'react-bootstrap';
import { Button, Input, SearchIcon } from '../../ui';

import { ErrorLoadCountersScreen } from '../errors/error-load-counters/ErrorLoadCountersScreen';
import { FooterScreen } from '../footer/FooterScreen';
import { ListCountersScreen } from '../list-counters/ListCountersScreen';
import { LoadingScreen } from '../loading/LoadingScreen';
import { NoCountersScreen } from '../no-counters/NoCountersScreen';

import './MainScreen.css';

export const MainScreen = (props) => {

    const [getCounters, setGetCounters] = useState({
        status: 0,
        data: []
    });
    const [loading, setLoading] = useState(true);
    const [selectedItems, setSelectedItems] = useState([]);
    const [searchValue, setSearchValue] = useState('');
    const [searchFocus, setSearchFocus] = useState(false);
    const [searchFilter, setSearchFilter] = useState([]);

    useEffect(() => {
        countersData();
    }, []);

    useEffect(() => {
        setSearchFilter(getCounters.data.filter(item => item.title.toLowerCase().includes(searchValue.toLocaleLowerCase())));
    }, [searchValue, getCounters]);

    const countersData = async () => {
        const data = await getAllCounters();
        setGetCounters(data);
        setTimeout(() => {
            setLoading(false);
        }, 1000);
    }

    const updateSelectedItems = (items) => {
        setSelectedItems(items);
        console.log({ main: selectedItems });
    }

    const handleChangeSearch = (e) => {
        e.preventDefault();
        setSearchValue(e.target.value);
    }

    const handleCancelSearch = () => {
        setSearchValue('');
        console.log(searchValue);
        setSearchFocus(false);
    }

    const handleOnFocus = () => {
        setSearchFocus(true);
    }
    
    return (
        <>
            <Container className="mh-100 py-4">
                <Row>
                    <Col className="d-flex align-items-center">
                        <div className="search-icon position-absolute px-3">
                            <SearchIcon />
                        </div>
                        <Input
                            className="search-input pl-5"
                            disabled={getCounters.data.length > 0 ? false : true}
                            placeholder="Search Counters"
                            onChange={handleChangeSearch}
                            onFocus={handleOnFocus}
                            value={searchValue} />
                        {
                            (getCounters.data.length > 0 && searchFocus) &&
                            (<Button
                                className="ml-3"
                                color="white"
                                onClick={handleCancelSearch}>
                                Cancel
                            </Button>)
                        }
                    </Col>
                </Row>
            
                {
                    loading ? <LoadingScreen classNameTitle={'py-8 vh-90'} /> :
                        <>
                            {
                                getCounters.status !== 200 ?
                                    <ErrorLoadCountersScreen countersData={countersData} /> :
                                    getCounters.data.length > 0 ?
                                        <ListCountersScreen
                                            products={searchFilter}
                                            searchFocus={searchFocus}
                                            selectedItems={selectedItems}
                                            updateSelectedItems={items => updateSelectedItems(items)}
                                        /> :
                                        <NoCountersScreen />
                            }
                        </>
                }

                <FooterScreen
                    countersData={countersData}
                    selectedItems={selectedItems}
                    updateSelectedItems={items => updateSelectedItems(items)} />
            </Container>
        </>
    )
}
