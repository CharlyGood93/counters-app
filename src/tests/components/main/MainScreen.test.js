import React from 'react';
import '@testing-library/jest-dom';

import { shallow } from 'enzyme';
import { MainScreen } from '../../../components/main/MainScreen';

describe('Test in <MainScreen />', () => {
    const wrapper = shallow(<MainScreen />);

    test('It should show the <MainScreen /> component correctly ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should simulate search bar input', () => {
        const input = wrapper.find('.search-input');
        const value = 'Milky Shake';
        input.simulate('change', { target: { value } });
    });
});
