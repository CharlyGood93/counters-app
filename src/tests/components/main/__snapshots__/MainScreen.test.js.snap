// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <MainScreen /> It should show the <MainScreen /> component correctly  1`] = `
<Fragment>
  <Container
    className="mh-100 py-4"
    fluid={false}
  >
    <Row
      noGutters={false}
    >
      <Col
        className="d-flex align-items-center searchbar pb-3"
      >
        <div
          className="search-icon position-absolute px-3"
        >
          <SearchIcon />
        </div>
        <Input
          className="search-input pl-5"
          disabled={true}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search Counters"
          value=""
        />
      </Col>
    </Row>
    <LoadingScreen
      classNameTitle="py-8 vh-90"
    />
    <FooterScreen
      countersData={[Function]}
      selectedItems={Array []}
      updateSelectedItems={[Function]}
    />
  </Container>
</Fragment>
`;
